shell.py

```
@@ -0,0 +1,246 @@
+        _report_shell_error("RuntimeError occurred in shell.py", persist=not args.dev_no_save, exception_context=exc.__context__, final_exception=exc, run_id=runner._run_id if runner else None)
+        return 1
+    except Exception as exc:  # pylint: disable=broad-exception-caught
+        _report_shell_error("Exception occurred in shell.py", persist=not args.dev_no_save, exception_context=exc.__context__, final_exception=exc, run_id=runner._run_id if runner else None)
+        return 1
+    finally:
+        # Explicitly close the psycopg pool to avoid thread finalization warnings on interpreter shutdown.
+        try:
+            if hasattr(DB, "instance") and hasattr(DB.instance, "_pool"):
+                DB.instance.shutdown()
+        except Exception:  # pylint: disable=broad-exception-caught
+            pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)